test/integration/scheduler_perf/util.go

```diff
@@ -166,20 +188,39 @@ func (*metricsCollector) run(ctx context.Context) {
 func (pc *metricsCollector) collect() []DataItem {
 	var dataItems []DataItem
 	for _, metric := range pc.Metrics {
-		dataItem := collectHistogram(metric, pc.labels)
-		if dataItem != nil {
-			dataItems = append(dataItems, *dataItem)
+		itemCollection := collectHistogram(metric, pc.labels)
+		if itemCollection != nil {
```

Comment:
we don't need the `!= nil` check anymore

Reply:
Done